### DISCUSSION TITLE
why concatenated string classnames don't work?

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#11631] [Help]

**Comment by adamwathan** [comment]
Hey! How this works and what you should do instead is all explained in the link you shared:

https://tailwindcss.com/docs/content-configuration#dynamic-class-names

You should also read the "Class detection in-depth" section.

For your use case I would just use inline styles using the `style` prop. Or if you really need it to be a class, use the `style` prop to set a CSS variable and reference the variable using a static Tailwind class:

```
<div
  style={{ '--bg-color': userBgColor, '--hover-bg-color': userHoverBgColor }}
  className="bg-[var(--bg-color)] hover:bg-[var(--hover-bg-color)]"
>
```

**Reply by wongjn** [reply]
It works in the Playcode example because it uses the [Tailwind Play CDN](https://tailwindcss.com/docs/installation/play-cdn) to produce Tailwind CSS rules at runtime. However please note:

> The Play CDN is designed for development purposes only, and is not the best choice for production.
